src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -11,6 +13,43 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.C
 {
     internal sealed partial class CPSProject : AbstractProject, IWorkspaceProjectContext
     {
+        /// <summary>
+        /// Holds the task with continuations to sequentially execute all the foreground affinitized actions on the foreground task scheduler.
+        /// More specifically, all the notifications to workspace hosts are executed on the foreground thread. However, the project system might make project state changes
+        /// and request notifications to workspace hosts on background thread. So we queue up all the notifications for project state changes onto this task and execute them on the foreground thread.
+        /// </summary>
+        private Task _foregroundTaskQueue = Task.CompletedTask;
+
+        /// <summary>
+        /// Controls access to task queue
+        /// </summary>
+        private readonly object _queueGate = new object();
+
+        private void ExecuteForegroundAction(Action action)
```

Comment:
Those were the final pieces of code in the project tracker that were yet to be converted into being free threaded - essentially all of the code invoking [ExecuteOrScheduleForegroundAffinitizedAction](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/VisualStudioProjectTracker.cs,151) was temporary until someone with better knowledge of this code base can make it free threaded (possibly post RTW).